sdk/core/core-http/src/credentials/accessTokenRefresher.ts

```diff
@@ -34,8 +35,18 @@ export class AccessTokenRefresher {
    * then returns the token.
    */
   private async getToken(options: GetTokenOptions): Promise<AccessToken | undefined> {
-    this.lastCalled = Date.now();
-    const token = await this.credential.getToken(this.scopes, options);
+    let token: AccessToken | null = null;
+    if (this.isTokenValid && this.isTokenValid()) {
```

Comment:
if this token is valid, we call `getToken`... but if the token isn't valid we call `getToken`... ? 

Wouldn't it make more sense to always call `getToken` and if it returns back something falsy, we keep retrying until the token is truthy?

Reply:
Resolving since I reverted this.